app/models/dialog_field_importer.rb

```diff
@@ -36,11 +36,20 @@ def set_category_for_tag_control(dialog_field, dialog_field_attributes)
 
   def adjust_category(opts)
     return nil if opts[:category_description].nil?
-    category = if opts[:category_id]
-                 Category.find(opts[:category_id])
-               elsif opts[:category_name]
-                 Category.find_by_name(opts[:category_name])
-               end
+    category = find_category(opts)
     category.try(:description) == opts[:category_description] ? category.try(:id).to_s : nil
   end
+
+  def find_category(opts)
+    if opts[:category_id]
+      begin
+        cat = Category.find(opts[:category_id])
+        cat.try(:name) == opts[:category_name] ? cat : nil
+      rescue ActiveRecord::RecordNotFound
+        Category.find_by_name(opts[:category_name])
+      end
+    elsif opts[:category_name]
+      Category.find_by_name(opts[:category_name])
+    end
+  end
```

Comment:
@branic Thanks for the PR, this is a great start.  I would advise against using error conditions as control flow.  I think a simple change to use `find_by` gets us what we want.  The other change I am proposing below is that if the category name does not match `opts[:category_name]` do the lookup by `opts[:category_name]` instead of returning `nil`.

Let me know what you think about changing the method to the following:

```ruby
def find_category(opts)
  if opts[:category_id]
    cat = Category.find_by(:id => opts[:category_id])
    return cat if cat.try(:name) == opts[:category_name]
  end
  Category.find_by_name(opts[:category_name])
end
```

Reply:
@gmcculloug I've changed the method as you suggested.